ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/TemplateConfigurationFactory.kt

```diff
@@ -19,8 +19,7 @@ internal object TemplateConfigurationFactory {
         val paywallTemplate = PaywallTemplate.fromId(paywallData.templateName)
             ?: throw IllegalArgumentException("Unknown template ${paywallData.templateName}")
 
-        val localizedConfiguration = paywallData.configForLocale(locale)
-            ?: error("No configuration found for locale $locale")
+        val localizedConfiguration = paywallData.localizedConfiguration
```

Comment:
This uses the application context locale. Which seems to work on device, but doesn't work on the Android studio preview. To fix that, we need to use the activity context to get the locale and calculate it on the fly. This is rearchitecturing things a bit so will work on this on a different PR.

Reply:
It's weird that this is ignoring the `locale` parameter though, which means that we could end up with inconsistent values here and when calling `ProcessedLocalizedConfiguration` inside of `PackageConfigurationFactory`, right?

Reply:
Yeah, sorry I posted this comment before discovering the issue was with how we get the localized configuration, which is fixed in #1267 . So please ignore this

Reply:
Makes sense. So you can revert this?

Reply:
I can, though Im changing it in #1267 again to this, since we perform some calculations there to use the list of locales there. So I think we can keep it as it is now. Lmk if you would prefer me to revert this to minify the changes

Reply:
Okay that's fine then 👍🏻